### PR TITLE
samples: modules: thrift: Add newlib filter to samples.yaml

### PR DESCRIPTION
**Pull request by galak**
Not all toolchains support newlib so tests that require newlib need to have a filter to we don't try and build those tests on those testcases.  Add the following to samples.yaml to handle the issue:

	filter: TOOLCHAIN_HAS_NEWLIB == 1

### CODE REVIEW

**Review by stephanosio — CHANGES_REQUESTED** [review]
Will block this until https://github.com/zephyrproject-rtos/zephyr/pull/54637 is merged since that PR introduces a new way of filtering by full libc availability.

**Review by cfriedt — DISMISSED** [review]
LGTM - reminds me to try out picolibc. I don't think it was as ready when we initially introduced this feature.

**Comment by keith-packard** [comment]
Does this test just need `REQUIRES_FULL_LIBCPP`? We'd need to add a prompt to that.

**Comment by cfriedt** [comment]
@galak - please revisit the pr mentioned above by Stephanos

**Comment by cfriedt** [comment]
@galak - minor changes needed here

File: samples/modules/thrift/hello/client/prj.conf [hunk 1]
```diff
@@ -1,12 +1,10 @@
 # CONFIG_LIB_CPLUSPLUS Dependencies
-CONFIG_NEWLIB_LIBC=y
-CONFIG_NEWLIB_LIBC_NANO=n
+CONFIG_REQUIRES_FULL_LIBCPP=y
```
Comment (cfriedt) [inline samples/modules/thrift/hello/client/prj.conf:2]: It might even be better to move this to `select REQUIRES_FULL_LIBCPP` in `modules/thrift/Kconfig`.

Comment (cfriedt) [reply samples/modules/thrift/hello/client/prj.conf:2]: I wish there were an ISO C code generator upstream (instead of the one that depends on glib)

**Comment by cfriedt** [comment]
LGTM other than the one suggestion above. Filters are good as well - that will speed up CI.

**Review by stephanosio — COMMENTED** [review]
https://github.com/zephyrproject-rtos/zephyr/pull/57445 has been merged. Needs a rebase.

**Comment by galak** [comment]
> #57445 has been merged. Needs a rebase.

rebased, removed DNM.

**Review by stephanosio — APPROVED** [review]

**Review by keith-packard — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Pull request merged**